src/util/diproche/diprocheInterface.ts

```diff
@@ -0,0 +1,97 @@
+import Issue from "../../issueHandling/issue";
+import { concatOneIssueList } from "../../issueHandling/issueMapping";
+import { getAllIssues } from "../../vocabularyChecker/detectWrongSyntax";
+import { UnexpectedError } from "./Errors";
+import modes from "./predicateList.json";
+
+export type Mode = "propositionalLogic" | "firstOrderPredicateLogic" | "test";
```

Comment:
A type will definitly work here too, but I think an enum would do better. The autocomplete function is more usable then the one for this type implementation. You don't have do handle strings which will indicate that this is not just randomly changeable and you can use a secondary value for them (if undefined they will just be 0, 1, 2 etc.)

Reply:
I see your point and will try the enum.